Revives "Adds sound effects to surgeries"

nmajask opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Continues #64285 by [maxspells](https://github.com/maxspells), which added sound effects to surgery steps. Major differences are that this uses vars to remove some of the repetitive code and some mechanical steps use the tool's use sound as the preop sound if applicable. Op sounds are lists currently in case someone in the future wants to add multiple different sounds, but I can replace them if preferred. First PR here, so I'm probably messing something up.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Staring at progress bars is boring. This will hopefully make performing surgeries more engaging for MDs and feel less monotonous.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: maxspells
expansion: Surgery steps play sound effects appropriate to the step
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

nmajask committed:
Adds surgical sounds

Rough implementation as I don't have too much time tonight.

Fikou commented on code/modules/surgery/surgery_step.dm:
```diff
@@ -9,6 +9,9 @@
 	var/silicons_obey_prob = FALSE
+	var/list/preop_sound //Sound played when the step is started
+	var/list/success_sound //Sound played if the step succeeded
+	var/list/falure_sound //Sound played if the step fails
```
this should be failure

Fikou replied:
also why are they lists when its always just 1

nmajask replied:
ah, one sec

nmajask replied:
> Op sounds are lists currently in case someone in the future wants to add multiple different sounds, but I can replace them.

nmajask committed:
Fixed falure_sound

Fixed a typo

bluetshirtguy commented:
Awful sounding for the ears my dude.

nmajask commented:
> Awful sounding for the ears my dude.

How so?

maxspells commented:
fun fact I sampled mac and cheese and twigs snapping to make some of these sound as gross as possible

nmajask committed:
Removed the lists

Removed the lists and added sounds to revival surgery, stuff will probably break

Arctrooper8 commented:
It's bad enough that surgery now has pain (which does nothing) and causes people to scream while giving big red text warnings that do nothing but clutter your chat box. 
Giving it sound too though? Surgery already has things to make it less "monotonous" they don't help, I doubt this will either.
You mention sounds making it more interesting but over time they'll become boring and monotonous too, "just another part of surgery people" will think when listening to the awful sounds again and again.
I apologize if I've been rude, I just feel that adding sounds to surgery will not benefit anyone and would personally hamper my enjoyment of playing MD.

Pickle-Coding commented:
Could reduce the noise of the sounds if the patient is unconcious, to encourage anesthetics.

maxspells commented:
> You mention sounds making it more interesting but over time they'll become boring and monotonous too, "just another part of surgery people" will think when listening to the awful sounds again and again.

There's sound effects for literally everything engineering uses tools for, from screwdriving something to wrenching it etc. I don't see any reason why surgery has to be completely silent.

GuillaumePrata commented:
Surgery sounds aren't for the doctor only, it's there to give a hint of what is going on to people around it.
You can't weld a wall or airlock without people around you knowing what you are doing and that is good, makes you be on your toes if you are doing some crime.
On the other hand you can fully revive your fellow traitor in a 1x1 hole in a wall and no one will have a hint.
Also as a bonus point, medbay's treatment room looks like a meatshop on most shifts, now it will sound like one too.

bluetshirtguy commented:
> > Awful sounding for the ears my dude.
> 
> How so?

Something about making ss13 more visceral isn't a good direction, we have people exploding and there's organs but the lack of detail and simple look to it all to make that stuff more funny then gross. It reminds me of when wounds would describe themselves as gory as possible because ??? people want to read that? who knows.

Fikou commented on code/modules/surgery/surgery_step.dm:
```diff
@@ -130,6 +142,11 @@
+/datum/surgery_step/proc/play_success_sound(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(!success_sound)
+		return
+	playsound(get_turf(target), pick(success_sound), 75, TRUE, falloff_exponent = 12, falloff_distance = 1)
```
youre picking from not a list

Fikou commented on code/modules/surgery/surgery_step.dm:
```diff
@@ -145,6 +162,11 @@
+/datum/surgery_step/proc/play_failure_sound(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(!failure_sound)
+		return
+	playsound(get_turf(target), pick(failure_sound), 75, TRUE, falloff_exponent = 12, falloff_distance = 1)
```
same here

nmajask committed:
Fixes the playsounds still using pick

nmajask committed:
Added sounds to more surgeries

Added sounds to four more surgeries

bruhlookatthisdood commented:
alright, good job.
now add a visual minigame where you do the surgery. make it realistic.
this is a threat.

bruhlookatthisdood commented:
alright, good job.
now add a visual minigame where you do the surgery. make it realistic.
this is a threat.
